src/sql-parser/src/parser.rs

```diff
@@ -2583,6 +2583,10 @@ impl<'a> Parser<'a> {
                 self.expect_keyword(INTERVAL)?;
                 CreateSourceOptionName::TimestampInterval
             }
+            ON => {
+                self.expect_keyword(DISK)?;
+                CreateSourceOptionName::OnDisk
+            }
```

Comment:
I'm sorry for not realizing this when I made the original suggestion, but I'm wondering if we should make the option `DISK [= true]` or `SCRATCH DISK [= true]`. Just in case we need to stabilize the option. WDYT? Does that read better to y'all than "on disk"? It does to my ear, but maybe that' sjust me.

Reply:
`DISK` seems good to me!